graph/topo/johnson_cycles.go

```diff
@@ -70,6 +70,114 @@ func DirectedCyclesIn(g graph.Directed) [][]graph.Node {
 	return j.result
 }
 
+// DirectedCyclesContaining returns the set of elementary cycles in the graph g containing the node with ID vid.
+// This is a slight deviation from Johnson's.
```

Comment:
The final sentence is not needed.

Reply:
> Thank you for sending this.

Sure, thanks for working on the package. It's really helpful and well-written.

> Can you provide a provenance for these additions?

I am not 100% sure what you exactly mean by provenance. I think the answer is this pull request (I authored the modifications). However, I got the idea for `DirectedCyclesOfMaxLen` from someone else's fork of this package.

> Also, I'm interested to know how you would intend using these. I have suspicions, but it would be nice to know.

My intended use case is for a high performance application where post-processing the output of `DirectedCyclesIn` is too slow.

Reply:
>I am not 100% sure what you exactly mean by provenance. I think the answer is this pull request (I authored the modifications). However, I got the idea for `DirectedCyclesOfMaxLen` from someone else's fork of this package.

Yeah. I was just wondering if there was a paper.

>My intended use case is for a high performance application where post-processing the output of `DirectedCyclesIn` is too slow.

If it's not sensitive, I was wondering what the use is. I see crypto and finance in your repos, so I was guessing in that context, either algotrading or blockchain analysis.